ironforgedbot/command_tree.py

```diff
@@ -0,0 +1,140 @@
+import logging
+
+import discord
+
+from ironforgedbot.commands.hiscore.cmd_score import cmd_score
+from ironforgedbot.commands.hiscore.cmd_breakdown import cmd_breakdown
+from ironforgedbot.commands.ingots.cmd_view_ingots import cmd_view_ingots
+from ironforgedbot.commands.ingots.cmd_add_ingots import cmd_add_ingots
+from ironforgedbot.commands.ingots.cmd_add_ingots_bulk import cmd_add_ingots_bulk
+from ironforgedbot.commands.ingots.cmd_update_ingots import cmd_update_ingots
+from ironforgedbot.commands.raffle.cmd_raffle_admin import cmd_raffle_admin
+from ironforgedbot.commands.raffle.cmd_raffle_buy_tickets import cmd_buy_raffle_tickets
+from ironforgedbot.commands.raffle.cmd_raffle_tickets import cmd_raffle_tickets
+from ironforgedbot.commands.admin.cmd_sync_members import cmd_sync_members
+from ironforgedbot.commands.admin.cmd_log import cmd_log
+from ironforgedbot.commands.roster.cmd_roster import cmd_roster
+
+from ironforgedbot.client import DiscordClient
+from ironforgedbot.common.responses import send_error_response
+
+logger = logging.getLogger(__name__)
+
+
+class IronForgedCommandTree(discord.app_commands.CommandTree):
+    async def on_error(
+        self,
+        interaction: discord.Interaction,
+        error: discord.app_commands.AppCommandError,
+    ):
+        if isinstance(error, discord.app_commands.CheckFailure):
+            logger.warning(error)
+
+            await interaction.response.defer(thinking=True, ephemeral=True)
```

Comment:
I'm not sure how on_error works here, but won't this iteraction be deferred by now? Or is discord.py provides new one?

Reply:
So currently, the only way `discord.app_commands.CheckFailure` can be raised is from the new decorator. So I've made sure that each `ironforgedbot/commands/cmd_**.py` file begins by deferring the response where relevant. 

So at this point we can safely assume it hasn't been deferred yet. I've also tested this a lot locally and can confirm.

The other interactions that end up here could have been deferred already depending on where the exception is raised. I didn't find a nice way to check if the interaction has already been deferred or not. Open to suggestions as the rest of this is a little brittle. Maybe just something to keep an eye on.